source/fasta/fasta.d

```diff
@@ -85,7 +87,7 @@ struct FastaToken
         result.fields = new B[fields.length];
         foreach(i, ref f; fields)
             result.fields[i] = f.value(buf);
-        result.sequence = sequence.value(buf);
+        result.sequence = sequence.value(buf).filter!(a =>!a.isWhite).to!string;
```

Comment:
On the second thought, I agree to revert this part. I will write a `fastaValidate` function, and a fastaCleanup function. What do you think?

Reply:
> I did not know that the filter will allocate memory on the heap.

It's not `filter`, but the `to!string` part. But `filter` is going to auto-decode the unicode, making it really slow, compared to something more specialized (that knows everything is ascii).

> On the second thought, I agree to revert this part. I will write a fastaValidate function, and a fastaCleanup function. What do you think?

I'm not sure what the algorithm that would process the data looks like, but if, for instance, it uses a `switch` on the character, you can just have a `default` label that ignores whitespace and anything else anyway. Filtering out the white-space before that algorithm is redundant. It would only be for printing or validation.

But having separate functions that provide what is needed in those specialized cases, as fast as possible, would be useful.